### PR TITLE
Add first round of File Types

### PR DESCRIPTION
**Pull request by MannyGrill**
+ c
+ cs
+ cpp
+ css
+ javascript
+ jsx
+ coffeescript
+ diff
+ go
+ graphql
+ handlebars
+ json
+ less
+ makefile
+ markdown
+ objectivec

### CODE REVIEW

File: src/fileTypes/perl-file.pl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/perl
+
+open(DATA, "<file.txt") or die "Couldn't open file file.txt, $!";
+
+while(<DATA>) {
+   print "$_";
+}
```
Comment (josh-jw) [inline src/fileTypes/perl-file.pl:7]: Perl comment

File: src/fileTypes/python-file.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+prices = {'apple': 0.40, 'banana': 0.50}
+my_purchase = {
+    'apple': 1,
+    'banana': 6}
+grocery_bill = sum(prices[fruit] * my_purchase[fruit]
+                   for fruit in my_purchase)
+print ('I owe the grocer $%.2f' % grocery_bill)
```
Comment (josh-jw) [inline src/fileTypes/python-file.py:7]: Python comment

File: src/fileTypes/ruby-file.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/ruby
+
+aFile = File.new("input.txt", "r")
+if aFile
+   content = aFile.sysread(20)
+   puts content
+else
+   puts "Unable to open file!"
+end
```
Comment (josh-jw) [inline src/fileTypes/ruby-file.rb:9]: Ruby comment

File: src/fileTypes/rust-file.rs [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+use std::io::BufReader;
+use std::path::Path;
+
+fn main() {
+    let path = Path::new("./textfile");
+    let mut file = BufReader::new(File::open(&path));
+    for line in file.lines() {
+        println!("{}", line);
+    }
+}
```
Comment (josh-jw) [inline src/fileTypes/rust-file.rs:11]: Rust comment

File: src/fileTypes/swift-file.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        try text.write(to: fileURL, atomically: false, encoding: .utf8)
+    }
+    catch {/* error handling here */}
+
+    //reading
+    do {
+        let text2 = try String(contentsOf: fileURL, encoding: .utf8)
+    }
+    catch {/* error handling here */}
+}
```
Comment (josh-jw) [inline src/fileTypes/swift-file.swift:20]: Swift comment

File: src/fileTypes/c-file.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+   for(i = 0; i < num; ++i)
+   {
+      printf("For student%d\nEnter name: ", i+1);
+      scanf("%s", name);
+
+      printf("Enter marks: ");
+      scanf("%d", &marks);
+
+      fprintf(fptr,"\nName: %s \nMarks=%d \n", name, marks);
+   }
```
Comment (MannyGrill) [inline src/fileTypes/c-file.c:27]: C comment

File: src/fileTypes/coffeescript-file.coffee [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+# Objects:
+math =
+  root:   Math.sqrt
+  square: square
+  cube:   (x) -> x * square x
+
+# Splats:
+race = (winner, runners...) ->
+  print winner, runners
```
Comment (MannyGrill) [inline src/fileTypes/coffeescript-file.coffee:22]: Coffee script comment

File: src/fileTypes/cplusplus-file.cpp [hunk 8]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+#include <fstream>
+using namespace std;
+
+int main() {
+  // Create and open a text file
+  ofstream MyFile("filename.txt");
+
+  // Write to the file
+  MyFile << "Files can be tricky, but it is fun enough!";
```
Comment (MannyGrill) [inline src/fileTypes/cplusplus-file.cpp:10]: C++ comment

File: src/fileTypes/csharp-file.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+namespace Microsoft.Samples.TimeZoneConvertCS {
+  /// TimeZoneCollection
+  /// implements a sorted list of TimeZones by reading the Windows 
+  /// Registry (there's currently no other way to do it)
+  public class TimeZoneList : System.Collections.ArrayList {
+    public TimeZoneList() {
+      //initialize the list
+      this.Initialize();  
+    }
+
```
Comment (MannyGrill) [inline src/fileTypes/csharp-file.cs:19]: C# comment

File: src/fileTypes/css-file.css [hunk 10]
```diff
@@ -0,0 +1,22 @@
+
+/* Applies to the entire body of the HTML document (except where overridden by more specific
+selectors). */
+body {
+    margin: 25px;
+    background-color: rgb(240,240,240);
+    font-family: arial, sans-serif;
+    font-size: 14px;
+  }
```
Comment (MannyGrill) [inline src/fileTypes/css-file.css:9]: CSS comment

File: src/fileTypes/diff-file.diff [hunk 11]
```diff
@@ -0,0 +1,19 @@
+--- lao	Sat Jan 26 23:30:39 1991
++++ tzu	Sat Jan 26 23:30:50 1991
+@@ -1,7 +1,6 @@
+-The Way that can be told of is not the eternal Way;
+-The name that can be named is not the eternal name.
+ The Nameless is the origin of Heaven and Earth;
+-The Named is the mother of all things.
++The named is the mother of all things.
++
+ Therefore let there always be non-being,
```
Comment (MannyGrill) [inline src/fileTypes/diff-file.diff:10]: diff comment

File: src/fileTypes/go-file.go [hunk 12]
```diff
@@ -0,0 +1,7 @@
+package main
+
+import "fmt"
+
+func main() {
+    fmt.Println("hello world")
+}
```
Comment (MannyGrill) [inline src/fileTypes/go-file.go:7]: go comment

File: src/fileTypes/graphql-file.graphql [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    id: ID!
+    # The tweet text. No more than 140 characters!
+    body: String
+    # When the tweet was published
+    date: Date
+    # Who published the tweet
+    Author: User
+    # Views, retweets, likes, etc
+    Stats: Stat
+}
```
Comment (MannyGrill) [inline src/fileTypes/graphql-file.graphql:11]: GraphQL comment

File: src/fileTypes/handlebar-file.handlebars [hunk 14]
```diff
@@ -0,0 +1,5 @@
+<script id="student-scores" type="text/x-handlebars-template">
+{{#studentScore this}}
+  <li>{{ firstName }} {{ lastName }}, your total score is <b>{{ score }}</b></li>
+{{/studentScore}}
+</script>
```
Comment (MannyGrill) [inline src/fileTypes/handlebar-file.handlebars:5]: Handlebars comment

File: src/fileTypes/javascript-file.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+{
+	hours1 = "0" + (hours - 12);
+strToAppend = "PM";
+}
+else if (hours <12)
+{
+	hours1 = "0" + hours;
+	strToAppend = "AM";
+}
+else
```
Comment (MannyGrill) [inline src/fileTypes/javascript-file.js:20]: JS comment

File: src/fileTypes/jsXML-file.jsx [hunk 16]
```diff
@@ -0,0 +1,3 @@
+const myelement = <h1>I Love JSX!</h1>;
+
+ReactDOM.render(myelement, document.getElementById('root'));
```
Comment (MannyGrill) [inline src/fileTypes/jsXML-file.jsx:3]: JSX comment

File: src/fileTypes/json-file.json [hunk 17]
```diff
@@ -0,0 +1,57 @@
+{
+    "colors": [
+      {
+        "color": "black",
+        "category": "hue",
+        "type": "primary",
+        "code": {
+          "rgba": [255,255,255,1],
+          "hex": "#000"
+        }
```
Comment (MannyGrill) [inline src/fileTypes/json-file.json:10]: JSON comment

File: src/fileTypes/less-file.less [hunk 18]
```diff
@@ -0,0 +1,8 @@
+@primarycolor: #FF7F50;  
+@color:#800080;  
+h2{  
+color: @primarycolor;  
+}  
+h3{  
+color: @color;  
+}
```
Comment (MannyGrill) [inline src/fileTypes/less-file.less:8]: LESS comment

File: src/fileTypes/make-file.mk [hunk 19]
```diff
@@ -0,0 +1,8 @@
+blah: blah.o
+    cc blah.o -o blah # Runs third
+
+blah.o: blah.c
+    cc -c blah.c -o blah.o # Runs second
+
+blah.c:
+    echo "int main() { return 0; }" > blah.c # Runs first
```
Comment (MannyGrill) [inline src/fileTypes/make-file.mk:8]: make file comment

File: src/fileTypes/markdown-file.md [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+
+Enable typographer option to see result.
+
+(c) (C) (r) (R) (tm) (TM) (p) (P) +-
+
+test.. test... test..... test?..... test!....
+
+!!!!!! ???? ,,  -- ---
+
+"Smartypants, double quotes" and 'single quotes'
```
Comment (MannyGrill) [inline src/fileTypes/markdown-file.md:28]: Markdown comment

File: src/fileTypes/objectivec-file.m [hunk 21]
```diff
@@ -0,0 +1,8 @@
+NSFileManager *filemgr;
+
+filemgr = [NSFileManager defaultManager];
+
+if ([filemgr isWritableFileAtPath: @"/tmp/myfile.txt"]  == YES)
+        NSLog (@"File is writable");
+else
+        NSLog (@"File is read only");
```
Comment (MannyGrill) [inline src/fileTypes/objectivec-file.m:8]: Objective C comment

**Pull request closed without merging**